backend/core/src/main/java/org/sonarsource/sonarlint/core/issue/IssueService.java

```diff
@@ -338,29 +339,42 @@ private void handleEvent(String connectionId, IssueChangedEvent event) {
   }
 
   private void republishPreviouslyRaisedIssues(String connectionId, IssueChangedEvent event) {
+    var isSonarCloud = serverApiProvider.getServerApiOrThrow(connectionId).isSonarCloud();
+    var isMQRMode = storageService.connection(connectionId).serverInfo().read(isSonarCloud).map(StoredServerInfo::isMQRMode).orElse(false);
```

Comment:
This is also the comment I posted on the other PR (sorry, it got merged because I saw no further comments). Since I switched to storing raw values, when reading it, I need a way to convert the protobuf enum DEFAULT/STANDARD/MQR to a simple boolean. Otherwise, the StoredServerInfo object will contain the protobuf type, which seems off. But then, the cons of computing it directly when reading is that I need the `isSonarCloud` information, perhaps it should be stored with it?

Reply:
What I would do:
* we usually try to keep in the storage an image of what's on the server. So I would store the boolean setting that you synchronized. Since it might be absent on older versions or SC, you can use an optional bool. 
* with the optional boolean it means you represent 3 states: unknown, standard or MQR. Identifying the exact mode to use is not the purpose of the storage class IMO, it's more the class that uses the storage class. So in this IssueService (or maybe you can introduce another class if you need to do it in several places) you can determine the real mode based on 3 things: the server kind, the server version and the stored setting

That's just a proposal. You can also keep an enum instead of the boolean if you prefer, I would just avoid passing isSonarCloud to the storage class

Reply:
I like your proposal, and I made the changes